components/Workspaces/TrackedRepoWizard/SearchByReposStep.tsx

```diff
@@ -30,18 +31,21 @@ const EmptyState = () => {
 export const SearchByReposStep = ({
   onSearch,
   onSelectRepo,
+  onToggleRepo,
   repositories,
   searchedRepos,
   suggestedRepos = [],
 }: SearchByReposStepProps) => {
-  const [filteredRepositories, setFilteredRepositories] = useState(repositories);
+  const [filteredRepositories, setFilteredRepositories] = useState<Map<string, boolean>>(repositories);
 
   const onFilterRepos = (search: string) => {
-    setFilteredRepositories(
-      repositories.filter((repo) => {
-        return repo.includes(search);
-      })
-    );
+    setFilteredRepositories((repositories) => {
+      return new Map(
```

Comment:
Not sure if there's a better to create a new `Map` in this scenario @brandonroberts.`

Reply:
I think its reasonable. I think it reads clearer the way you did it further down

```ts
const updates = [...repositories(...etc)];

return new Map(updates);
```